code/modules/modular_computers/file_system/programs/virtual_pet.dm

```diff
@@ -0,0 +1,569 @@
+			pet.ai_controller?.set_blackboard_key(BB_LIGHTS_ABILITY, lights)
+		if(3)
+			var/datum/action/cooldown/mob_cooldown/capture_photo/photo_ability = new(pet)
+			photo_ability.Grant(pet)
+			pet.ai_controller?.set_blackboard_key(BB_PHOTO_ABILITY, photo_ability)
+			RegisterSignal(photo_ability.ability_camera, COMSIG_CAMERA_IMAGE_CAPTURED, PROC_REF(on_photo_captured))
+
+/datum/computer_file/program/virtual_pet/proc/announce_global_updates(message)
+	if(isnull(message))
+		return
+	var/list/message_to_announce = list(
+		"name" = pet.name,
+		"pet_picture" =  icon2base64(profile_picture),
+		"message" = message,
+		"likers" = list(REF(src))
+	)
+	if(length(GLOB.global_pet_updates) >= MAX_UPDATE_LENGTH)
+		GLOB.global_pet_updates.Cut(1,2)
+
+	GLOB.global_pet_updates += list(message_to_announce)
+	playsound(computer.loc, 'sound/items/orbie_notification_sound.ogg', 50)
+
+/datum/computer_file/program/virtual_pet/proc/remove_pet(datum/source)
+	SIGNAL_HANDLER
+	pet = null
+	computer.remove_file(src) //all is lost we no longer have a reason to exist
+
+/datum/computer_file/program/virtual_pet/kill_program(mob/user)
+	if(pet.loc != computer)
+		pet.forceMove(computer) //recall the hologram back to the pda
+	STOP_PROCESSING(SSprocessing, src)
+	return ..()
+
+/datum/computer_file/program/virtual_pet/proc/get_pet_state()
+	if(isnull(pet))
+		return
+
+	if(pet.loc == computer)
+		return PET_STATE_ASLEEP
+
+	if(happiness/max_happiness > 0.8)
+		return PET_STATE_HAPPY
+
+	if(hunger/max_hunger < 0.5)
+		return PET_STATE_HUNGRY
+
+	return PET_STATE_NEUTRAL
+
+/datum/computer_file/program/virtual_pet/ui_data(mob/user)
+	var/list/data = list()
+	data["currently_summoned"] = (pet.loc != computer)
+	data["selected_area"] = (selected_area ? selected_area.name : "No location set")
+	data["pet_state"] = get_pet_state()
+	data["hunger"] = hunger
+	data["maximum_hunger"] = max_hunger
+	data["pet_hat"] = (length(selected_hat) ? selected_hat["name"] : "none")
+	data["can_reroll"] = COOLDOWN_FINISHED(src, area_reroll)
+	data["can_summon"] = COOLDOWN_FINISHED(src, summon_cooldown)
+	data["can_alter_appearance"] = COOLDOWN_FINISHED(src, alter_appearance_cooldown)
+	data["pet_name"] = pet.name
+	data["steps_counter"] = steps_counter
+	data["in_dropzone"] = (istype(get_area(computer), selected_area))
+	data["pet_area"] = (pet.loc != computer ? get_area_name(pet) : "Sleeping in PDA")
+	data["current_exp"] = current_level_progress
+	data["required_exp"] = to_next_level
+	data["happiness"] = happiness
+	data["maximum_happiness"] = max_happiness
+	data["level"] = level
+	data["pet_color"] = ""
+
+	var/color_value = LAZYACCESS(pet.atom_colours, FIXED_COLOUR_PRIORITY)
+	for(var/index in possible_colors)
+		if(possible_colors[index] == color_value)
+			data["pet_color"] = index
+			break
+
+	data["pet_gender"] = ""
+	switch(pet.gender)
+		if(MALE)
+			data["pet_gender"] = "male"
+		if(FEMALE)
+			data["pet_gender"] = "female"
+		if(NEUTER)
+			data["pet_gender"] = "neutral"
```

Comment:
```suggestion
	data["pet_gender"] = pet.gender
```
You also don't take into account plural here

Reply:
the ui doesnt have enough space for 4 buttons so im just limiting it to these 3